packages/components/music/UploadMusicModal/UploadTrack.tsx

```diff
@@ -78,23 +85,27 @@ export const UploadTrack: React.FC<Props> = ({ onUploadDone }) => {
       await makePost(JSON.stringify(trackWithCreationDate));
     } catch (err) {
       console.error("post submit err", err);
+      setIsUploadLoading(false);
+      setIsProgressBarShown(false);
       setToastError({
         title: "Post creation failed",
         message: err instanceof Error ? err.message : `${err}`,
       });
     }
-    setIsUploading(false);
   };
 
   const onPressUpload = async () => {
-    setIsLoading(true);
+    setIsUploadLoading(true);
+    setIsProgressBarShown(true);
     if (
       !selectedWallet?.connected ||
       !selectedWallet.address ||
       !localAudioFile
     ) {
       return;
     }
+    setStep(feedPostingStep("GENERATING_KEY"));
```

Comment:
Is it not better to define an enum ?
To have values at once place, and if we want to change the step `GENERATING_KEY` to `GENERATING_KEY_AND_WHATEVER` we have only one line to change and not every occurrences of the step

Reply:
After discuss, I replaced by an `enum`. It's easier to modify yes !